proxy/reverse.go

```diff
@@ -92,6 +94,11 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	removeSingleHopHeaders(&proxyreq.Header)
 	maybeSetForwardedFor(proxyreq)
 
+	if err := checkProxyLoop(proxyreq, p.proxyID, 10); err != nil {
+		plog.Debugf("found proxy loop (%v)", err)
+		return
```

Comment:
As suggested [here](https://github.com/coreos/etcd/pull/3636#issuecomment-145682590), if it finds a loop, it **returns** without updating endpoints.